API-9946: add API lifecycle headers. (#1052)
livechat/livechat-public-docs: src/pages/messaging/agent-chat-api/v3.4/index.mdx

## Versioning

This document describes the **Agent Chat Web API v3.4**. This is the developer preview version that provides a preview of the upcoming changes to the API. **It's not open to public use.** However, if you want to test some features, contact us at developers@livechat.com or <a href="#open-chat" onClick={openChatWindow}>ask on the chat</a>, and we'll give you access. [Read more about versioning...](/messaging/#api-versioning)

### Lifecycle headers

> API responses might contain one of the two headers related to the version's lifcycle, `Legacy` and `Deprecation`. Each of these headers contains the date when the associated stage ends.

Reviewer: ```suggestion
API responses will contain one of two headers related to the API lifecycle: `Legacy` or `Deprecation`. These headers specify when the associated stage ends (in the `YYYY-MM-DD` format).
```
